builtin/merge-ours.c

```diff
@@ -11,12 +11,24 @@
 #include "git-compat-util.h"
 #include "builtin.h"
 #include "diff.h"
+#include "parse-options.h"
+
+static const char * const merge_ours_usage[] = {
+	N_("git merge_ours [arg]..."),
+	NULL,
+};
 
 static const char builtin_merge_ours_usage[] =
 	"git merge-ours <base>... -- HEAD <remote>...";
 
 int cmd_merge_ours(int argc, const char **argv, const char *prefix)
 {
+	struct option options[] = {
+		OPT_END()
+	};
+
+	argc = parse_options(argc, argv, prefix, options, merge_ours_usage, 0);
+
 	if (argc == 2 && !strcmp(argv[1], "-h"))
 		usage(builtin_merge_ours_usage);
```

Comment:
I think this is now incorrect. The expected number of arguments after option parsing is `0` (except if you want to keep `argv[0]`, which in this case you don't).

Besides, `parse_options()` already handles the `-h` option...

Reply:
I'm actually a little confused, dscho, by what's going on here. Why does the usage indicate that all these options can be given, when the command actually just checks current state and ignores whatever options came in? Is the option list standard across all merge strategies?

Reply:
> Why does the usage indicate that all these options can be given

Which options :-)

You mean all those arguments? Because that is the way the merge strategies work: they receive (apart from options, _if_ the merge strategy can handle any) the merge base(s), a double-dash, and then the merge heads, starting with `HEAD`.

> when the command actually just checks current state and ignores whatever options came in?

That's what `ours` does: it ignores all incoming merge heads and simply chooses what is in `HEAD` already.

Technically, it _should_ verify that the first argument after the `--` actually references `HEAD`, but practically, it does not make any difference because the command-line is generated by Git itself and it always starts with `HEAD`.

> Is the option list standard across all merge strategies?

Yes. Otherwise, there would be little point in having a `try_merge_command()` function that takes the name of the merge strategy as a parameter.

Reply:
> I think this is now incorrect. The expected number of arguments after option parsing is `0` (except if you want to keep `argv[0]`, which in this case you don't).
> 
> Besides, `parse_options()` already handles the `-h` option...

Ah All right, I will keep this also in mind when doing my changes! Thank you!